agenthub/codeact_agent/codeact_agent.py

```diff
@@ -222,7 +222,13 @@ def _get_messages(self, state: State) -> list[dict[str, str]]:
 
             # add regular message
             if message:
-                messages.append(message)
+                # handle error if the message is the SAME role as the previous message
+                # litellm.exceptions.BadRequestError: litellm.BadRequestError: OpenAIException - Error code: 400 - {'detail': 'Only supports u/a/u/a/u...'}
+                # there should not have two consecutive messages from the same role
+                if messages and messages[-1]['role'] == message['role']:
```

Comment:
I didn't test, but it looks like this will also merge the example message (in_context_sample) with the first user message. If that's the case, will it have some effect on evals, since the first user message is the task?

Reply:
I think the original implementation of CodeAct is actually merging this two message together - then it get separated at some moment. I think we had a "-- end of example --" message there, so i think we should be fine?

Reply:
Definitely possible, it _looks_ separated, and we have also a hard-coded "NOW, LET'S START!" after that.

Sorry, I'm running SWE-bench at the moment on another branch, as soon as it's done I can try this a little bit. But I think merging this is okay before, it doesn't have to wait. This is a small thing we can tweak if needed, IMHO.